src/gpu/intel/jit/ir/tensor.hpp

```diff
@@ -1012,6 +808,78 @@ class layout_t {
     std::vector<layout_block_t> blocks_;
 };
 
+inline dim_t inner_block(const layout_t &layout, const pvar_t &dim,
+        bool skip_outer = true, bool inner_only = true) {
+    std::vector<dim_t> dim_blocks;
+    for (auto &b : layout.blocks()) {
+        if (b.dim == dim) dim_blocks.push_back(b.block);
+    }
+    dim_t ret = 1;
+    int nblocks = (int)dim_blocks.size();
+    int lo = 0;
+    int hi = skip_outer ? nblocks - 1 : nblocks;
+    if (inner_only) hi = std::min(hi, 1);
+    for (int i = lo; i < hi; i++)
+        ret *= dim_blocks[i];
+    return ret;
+}
+
+// Storage size in bytes.
+inline dim_t size_bytes(const layout_t &layout, dim_t alignment = 1) {
+    if (layout.is_empty()) return 0;
+    dim_t max_off = 0;
+    dim_t max_block_size = 0;
+    for (auto &b : layout.blocks()) {
+        max_off += (b.block - 1) * (dim_t)b.stride;
+        max_block_size = std::max(max_block_size, b.block * (dim_t)b.stride);
+    }
+    dim_t max_elems = std::max(max_off + 1, max_block_size);
+    return utils::rnd_up(
+            max_elems * layout.type().size() / layout.type().packing(),
+            alignment);
+}
+
+template <typename T = expr_t>
+T offset_in_bytes(const layout_t &layout, const coord_t &coord = {},
```

Comment:
To align with `size_bytes()`:
```suggestion
T offset_bytes(const layout_t &layout, const coord_t &coord = {},
```